test/e2e/cluster.go

```diff
@@ -0,0 +1,203 @@
+		ctx := context.Background()
+		err := createPVC(ctx, clients.Kubernetes)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = createPodWithPVC(ctx, clients.Kubernetes)
+		Expect(err).NotTo(HaveOccurred())
+
+		Eventually(func() error {
+			return verifyPodSucceeded(clients.Kubernetes)
+		}).Should(BeNil())
+	})
+
+	Specify("Can create load balancer services", func() {
+		ctx := context.Background()
+		err := createLoadBalancerService(ctx, clients.Kubernetes, "elb", map[string]string{})
+		Expect(err).NotTo(HaveOccurred())
+
+		err = createLoadBalancerService(ctx, clients.Kubernetes, "ilb", map[string]string{
+			"service.beta.kubernetes.io/azure-load-balancer-internal": "true",
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		Eventually(func() error {
+			return verifyServiceIsReady(ctx, clients.Kubernetes, "elb")
+		}, 5*time.Minute, 10*time.Second).Should(BeNil())
+
+		Eventually(func() error {
+			return verifyServiceIsReady(ctx, clients.Kubernetes, "ilb")
+		}, 5*time.Minute, 10*time.Second).Should(BeNil())
+	})
+})
+
+func createPVC(ctx context.Context, cli kubernetes.Interface) error {
+	pvcStorage, err := resource.ParseQuantity("2Gi")
+	if err != nil {
+		return err
+	}
+	pvcName := fmt.Sprintf("%s-pvc", testPodName)
+	storageClass := "managed-premium"
+	_, err = cli.CoreV1().PersistentVolumeClaims(testNamespace).Create(&corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: pvcName,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				corev1.PersistentVolumeAccessMode("ReadWriteOnce"),
+			},
+			StorageClassName: to.StringPtr(storageClass),
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: pvcStorage,
+				},
+			},
+		},
+	})
+	return err
+}
+
+func createPodWithPVC(ctx context.Context, cli kubernetes.Interface) error {
+	pvcName := fmt.Sprintf("%s-pvc", testPodName)
+	volumeName := fmt.Sprintf("%s-vol", pvcName)
+	_, err := cli.CoreV1().Pods(testNamespace).Create(&corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testPodName,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  testPodName,
+					Image: testPodName,
+					Command: []string{
+						"/bin/dd",
+						"if=/dev/urandom",
+						fmt.Sprintf("of=/data/%s.bin", testNamespace),
+						"bs=1M",
+						"count=100",
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      volumeName,
+							MountPath: "/data",
+							ReadOnly:  false,
+						},
+					},
+				},
+			},
+			Volumes: []corev1.Volume{
+				{
+					Name: volumeName,
+					VolumeSource: corev1.VolumeSource{
+						PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+							ClaimName: pvcName,
+						},
+					},
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyNever,
+		},
+	})
+	return err
+}
+
+func verifyPodSucceeded(cli kubernetes.Interface) error {
```

Comment:
https://github.com/Azure/ARO-RP/blob/master/test/util/ready/ready.go#L23 ?

Reply:
@asalkeld this pod finishes quickly so we risk to miss "ready" state. That's why it is checking for Succeeded instead